### ISSUE TITLE
[Bug] Custom series doesn't remove previously rendered items any more

### ISSUE THREAD

**Issue by MPetak** [apache/echarts#16503]
### Version

5.3.0

### Link to Minimal Reproduction

https://codepen.io/m5ak/pen/gOXxBxw

### Steps to Reproduce

We render elements within custom series and when we try to remove some item from chart (for example data:[]) then echarts throws next exception: Uncaught TypeError: Cannot read properties of undefined (reading '__ec_inner_26'),
and items are not removed form chart. 

### Current Behavior

We render elements within custom series and when we try to remove some item from chart (for example data:[]) then echarts throws next exception: Uncaught TypeError: Cannot read properties of undefined (reading '__ec_inner_26'),
and items are not removed form chart. Items are just stacked one to another. 

### Expected Behavior

Expected is as in previous versions of echarts that element is removed from chart when data item of element is not defined.

### Environment

```markdown
- OS:Windows 10 Enterprise 20H2
- Browser: Google Chrome 98.0.4758.82
- Framework:
```

**Comment by rundleman** [comment]
Hello! I have the same problem. I am also very surprised by the performance increase in version 5.3.0. It is possible to manually delete old data. For example like this:
`  chart.setOption({ series: [] }, {
    notMerge: true,
  });`
https://codepen.io/rundleman/pen/yLPPzmj

**Comment by MPetak** [comment]
@rundleman Thank you for your response, but that is not solution for then issue, it is just workaround which is unfortunately not acceptable for me. We have downgraded Echarts to version 5.1.1 and I hope that I'll get some response from Echarts owners.

**Comment by MPetak** [comment]
As far as I know, in newer versions this bug still persists. 
Because of that we have downgraded ECharts to 5.2.1

**Comment by anymos** [comment]
finger crossed that it is working in the 5.2.1 ... without too much performance issues

**Comment by anymos** [comment]
@[susiwen8](https://github.com/susiwen8) I saw that you hve closed another issue yesterday, is there a way to contact the core team member for that issue or to get some pointer so I can participate and propose a PR with the fix ? thank you in advance !

**Comment by anymos** [comment]
Even by using chart.setOption({ series: [] }, { notMerge: true, }); the bug is still present with a filter:'none'... not too sure how to fix that

**Comment by anymos** [comment]
I tried to create a new bug from scratch to get more traction .. still no answer on this...

**Comment by MPetak** [comment]
It is resolved with version 5.4.0
